### ISSUE TITLE
False positive logging-format-truncated using percentage in the message without args

### ISSUE THREAD

**Issue by moylop260** [pylint-dev/pylint#10458]
### Bug description

```python
import logging

logging.info('Good percent %')
```

### Configuration

```ini

```

### Command used

```shell
pylint --logging-format-style=old -d all -e logging-format-truncated --rcfile=/dev/null my_file.py
```

NOTE: Using `--logging-format-style=new` format is not reproduced

### Pylint output

```python
E1201: Logging format string ends in middle of conversion specifier (logging-format-truncated)
```

### Expected behavior

The percentage value is valid and it doesn't need to be scaped if there are not args to parse in the string

Notice the following example where the value is showed correctly without scaping vs scaping the value:
 - <img width="854" height="178" alt="Image" src="https://github.com/user-attachments/assets/d7ecc54f-2559-42f4-a3b6-1dc7264a1419" />

### Pylint version

```shell
pylint 4.0.0-dev0
astroid 4.0.0a0
Python 3.12.11 (main, Jun  3 2025, 15:41:47) [Clang 17.0.0 (clang-1700.0.13.3)]
```

### OS / Environment

MacOSX

### Additional dependencies

```python

```

**Comment by daltonbohning** [comment]
It seems v4.0.0 was released with this issue still present. As far as I can tell, the only workaround is to disable this check?

**Comment by Pierre-Sassoulas** [comment]
You can disable it locally when the % is legit with ``# pylint: disable-next=logging-format-truncated``. If you want to "fix" this I guess it shouldn't raise when there's no argument to be processed ? i.e. ``logging.info('Bad percent %', arg)`` raise but not ``logging.info('Good percent %')``. But then it wouldn't catch double errors, so I don't know, maybe the false positive is acceptable, what do you think ? Personally I use the new logging format.

**Comment by daltonbohning** [comment]
I would hope we don't have to disable it, since the `%` is valid when there are no arguments. When there are no arguments, all `%` are treated literally, but if there is at least one argument then all `%` must be valid format specifiers or must be escaped like `%%`.

What new logging format are you referring to?

**Comment by Pierre-Sassoulas** [comment]
The decision was taken in https://github.com/pylint-dev/pylint/issues/9999#issuecomment-2460083265, and implemented in https://github.com/pylint-dev/pylint/pull/10321, we caught 4 false negatives and no false positives in the primer at the time https://github.com/pylint-dev/pylint/pull/10321#issuecomment-2764557547.

I meant ``--logging-format-style=new`` (using f-strings in logging, the perf impact of doing this is negligible)

**Comment by daltonbohning** [comment]
Okay, I see. It seems the real problem lies with how python handles this, so the linter can never be perfect. I can live with this - thanks!
